internal/management/controller/database_controller_test.go

```diff
@@ -0,0 +1,311 @@
+			},
+		}
+		database = &apiv1.Database{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:       "db-one",
+				Namespace:  "default",
+				Generation: 1,
+			},
+			Spec: apiv1.DatabaseSpec{
+				ClusterRef: corev1.LocalObjectReference{
+					Name: cluster.Name,
+				},
+				Name:  "db-one",
+				Owner: "app",
+			},
+		}
+
+		db, dbMock, err = sqlmock.New(sqlmock.QueryMatcherOption(sqlmock.QueryMatcherEqual))
+		Expect(err).ToNot(HaveOccurred())
+
+		pgInstance := &postgres.Instance{
+			Namespace:   "default",
+			PodName:     "cluster-example-1",
+			ClusterName: "cluster-example",
+		}
+
+		f := fakeInstanceData{
+			instanceData: instanceData{instance: pgInstance},
+			db:           db,
+		}
+
+		fakeClient = fake.NewClientBuilder().WithScheme(schemeBuilder.BuildWithAllKnownScheme()).
+			WithObjects(cluster, database).
+			WithStatusSubresource(&apiv1.Cluster{}, &apiv1.Database{}).
+			Build()
+
+		r = &DatabaseReconciler{
+			Client:   fakeClient,
+			Scheme:   schemeBuilder.BuildWithAllKnownScheme(),
+			instance: &f,
+		}
+	})
+
+	AfterEach(func() {
+		Expect(dbMock.ExpectationsWereMet()).To(Succeed())
+	})
+
+	It("database object inherits error after patching", func(ctx SpecContext) {
+		// Mocking DetectDB
+		expectedValue := sqlmock.NewRows([]string{""}).AddRow("1")
+		dbMock.ExpectQuery(`SELECT count(*)
+		FROM pg_database
+		WHERE datname = $1`).WithArgs(database.Spec.Name).WillReturnRows(expectedValue)
+
+		// Mocking Alter Database
+		expectedError := fmt.Errorf("no permission")
+		expectedQuery := fmt.Sprintf("ALTER DATABASE %s OWNER TO %s",
+			pgx.Identifier{database.Spec.Name}.Sanitize(),
+			pgx.Identifier{database.Spec.Owner}.Sanitize(),
+		)
+		dbMock.ExpectExec(expectedQuery).WillReturnError(expectedError)
+
+		_, err := r.Reconcile(ctx, ctrl.Request{NamespacedName: types.NamespacedName{
+			Namespace: database.Namespace,
+			Name:      database.Spec.Name,
+		}})
+		Expect(err).ToNot(HaveOccurred())
+
+		var updatedDatabase apiv1.Database
+		err = fakeClient.Get(ctx, client.ObjectKey{
+			Namespace: database.Namespace,
+			Name:      database.Name,
+		}, &updatedDatabase)
+		Expect(err).ToNot(HaveOccurred())
+
+		Expect(updatedDatabase.Status.Ready).Should(BeFalse())
+		Expect(updatedDatabase.Status.Error).Should(ContainSubstring(expectedError.Error()))
+	})
+
+	It("properly marks the status on a succeeded reconciliation", func(ctx SpecContext) {
+		_, err := r.succeededReconciliation(ctx, database)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(database.Status.Ready).To(BeTrue())
+		Expect(database.Status.Error).To(BeEmpty())
+	})
+
+	It("properly marks the status on a failed reconciliation", func(ctx SpecContext) {
+		exampleError := fmt.Errorf("sample error for database %s", database.Spec.Name)
+
+		_, err := r.failedReconciliation(ctx, database, exampleError)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(database.Status.Ready).To(BeFalse())
+		Expect(database.Status.Error).To(BeEquivalentTo(exampleError.Error()))
+	})
```

Comment:
Now that we can test the `r.Reconcile` function, I think these two tests are not very significant.